consensus/blake3pow/consensus.go

```diff
@@ -625,25 +628,39 @@ func (blake3pow *Blake3pow) Finalize(chain consensus.ChainHeaderReader, header *
 			}
 		}
 		addressOutpointMap := make(map[string]map[string]*types.OutpointAndDenomination)
-		core.AddGenesisUtxos(state, nodeLocation, addressOutpointMap, blake3pow.logger)
+		core.AddGenesisUtxos(chain.Database(), &utxosCreate, nodeLocation, addressOutpointMap, blake3pow.logger)
 		if chain.Config().IndexAddressUtxos {
 			chain.WriteAddressOutpoints(addressOutpointMap)
 			blake3pow.logger.Info("Indexed genesis utxos")
 		}
+	} else {
+		multiSet = rawdb.ReadMultiSet(chain.Database(), header.ParentHash(nodeCtx))
+	}
+	if multiSet == nil {
+		blake3pow.logger.Fatalf("Multiset is nil for block %s", header.ParentHash(nodeCtx).String())
```

Comment:
you should probably return error here if the multiset is nil, in the progpow case as well, we will panic otherwise